modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -257,6 +260,324 @@ class AndroidMediaNdkCapture : public IVideoCapture
     }
 };
 
+
+
+class AndroidMediaNdkVideoWriter CV_FINAL :
+    public cv::IVideoWriter
+{
+    typedef struct {
+        int fourcc;
+        const char* mime;
+        OutputFormat muxerFormat;
+    }
+    FourCCInfo;
+
+    static const int64_t TIMEOUT = 1000L;
+    static const FourCCInfo FOURCC_INFO[];
+
+    static const FourCCInfo* findInfo(int fourcc) {
+        for( const FourCCInfo *it = FOURCC_INFO; NULL != it->mime; it++ ) {
+            if (fourcc == it->fourcc) return it;
+        }
+        return NULL;
+    }
+
+    AMediaFormat* format;
+    AMediaCodec* encoder;
+    AMediaMuxer* muxer;
+
+    #if __ANDROID_API__ >= 26
+    ANativeWindow* surface;
+    #endif
+
+    long frameIndex;
+    int width;
+    int height;
+    double frameRate;
+    ssize_t videoTrackIndex;
+    int fd;
+
+    void drainEncoder(bool end) {
+        if (end) {
+            #if __ANDROID_API__ >= 26
+            AMediaCodec_signalEndOfInputStream(encoder);
+            #else
+            writeBytes(NULL, 0);
+            #endif
+        }
+
+        AMediaCodecBufferInfo bufferInfo;
+        ssize_t bufferIndex;
+        size_t  bufferSize;
+        uint8_t *buffer;
+
+        while (true) {
+            bufferIndex = AMediaCodec_dequeueOutputBuffer(encoder, &bufferInfo, TIMEOUT);
+            if (bufferIndex >= 0) {
+                buffer = AMediaCodec_getOutputBuffer(encoder, (size_t)bufferIndex, &bufferSize);
+                if (NULL == buffer || 0 == bufferSize) break;
+                bufferInfo.presentationTimeUs = frameIndex * 1000000L / frameRate;
+                LOGV("Muxer write to track %d: %d byte(s)", (int)videoTrackIndex, (int)bufferInfo.size);
+                if (videoTrackIndex >= 0) AMediaMuxer_writeSampleData(muxer, (size_t)videoTrackIndex, buffer, &bufferInfo);
```

Comment:
I propose to move "then" section to the next line with indentation everywhere. It makes code more readable.

Reply:
Totaly agree